Dockerfile

```diff
@@ -1,37 +1,30 @@
-FROM golang:1.21 as build-stage
+# syntax=docker/dockerfile:1
```

Comment:
I think this is a fairly minimal/idiomatic modern Dockerfile, but some of the features like `--mount=type=cache` for caching and using `--platform=${BUILDPLATFORM}` for cross-compiling were new to me. The Docker documentation covers these all well, and this is pretty close to their multi-stage and multi-platform examples for Go but it may still be worth adding a bit more commentary to this file.

Reply:
+1 on more commentary 🤓 - i can guess at what those args do, but understanding some nuance of how they interact with the GH actions environment, if anything would need to change for a local dev environment, etc. would be helpful

Reply:
Added comments to clarify.